### PR TITLE
fix: fix regression caused by join on trials view

### PR DESCRIPTION
**Pull request by salonig23**
## Description

Join on trials view was causing regression, and slowing down GetExperiments. Fixed it by removing join and adding column to runs table. 

## Test Plan

Make sure all tests pass

For CLI - 
Experiment:
1. Create an experiment.
2. Wait for the experiment to complete.
3. Run `det e set log-retention-days <exp-id> --days 50`
4. Check the runs table in the database to see if the associated trials for the experiment have updated the log retention days to 50.
5.  Run `det e set log-retention-days <exp-id> --days none --forever`
6. Check the tasks table in the database to see if the associated trials for the experiment have updated the log retention days to -1.

Trial:
1. Create an experiment.
2. Wait for the experiment to complete.
3. Run `det t set log-retention-days <trial-id> --days 50` for any trial in the experiment.
4. Check the runs table in the database to see if the trial has updated the log retention days to 50.
5.  Run `det t set log-retention-days <trial-id> --days none --forever`
6. Check the tasks table in the database to see if the trial has updated the log retention days to -1.

For API - 
1. Create an experiment.
2. Wait for the experiment to complete.
3. Use the following API endpoint:

> curl -X 'PUT' \
>   'http://localhost:8080/api/v1/experiments/1/retain_logs' \
>   -H 'accept: application/json' \
>   -H 'Content-Type: application/json' \
>   -d '{
>   "experimentId": 1,
>   "numDays": 40
> }'

Check the tasks table in the database to see if the associated trials for the experiment have updated the log retention days to 40.

> curl -X 'PUT' \
>  'http://localhost:8080/api/v1/trials/1/retain_logs' \
>   -H 'accept: application/json' \
>   -H 'Content-Type: application/json' \
>   -d '{
>   "trialId": 1,
>   "numDays": -1
> }'

Check the tasks table in the database to see if the trial has updated the log retention days to -1.

For WebUI -
1. Create a few experiments.
2. Go to the Experiments list in the Web UI and select a few of them and then click on the actions button.
EC0E20C0-1338-4317-BDCC-F7F637E8A2D7
3. Select the Retain Logs option from the drop down.
4. Type any number in the Number of days field when prompted.
8. Wait for a success message.
9. Check the runs table in the database to see if the associated trials for the experiment have updated the log retention days to the number you put in.
10. Select a few more experiments and follow the above steps, but this time click on the 'Forever' checkbox instead.
11. Wait for a success message.
12.  Check the runs table in the database to see if the associated trials for the experiment have updated the log retention days to -1.
13. Also check the Trial List Table to see if the Log Retention Days Column shows up and has the correct days.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by NicholasBlaskey — APPROVED** [review]
Looks good to me

I would check the performance test run first and get someone who has more context on this project to make sure it makes sense to have it on both tasks and runs

File: proto/src/determined/run/v1/run.proto [hunk 1]
```diff
@@ -109,4 +109,6 @@ message FlatRun {
   bool parent_archived = 18;
   // Data related the the experiment associated with this run.
   optional FlatRunExperiment experiment = 19;
+  // Number of days to retain logs for.
+  optional int32 log_retention_days = 20;
```
Comment (NicholasBlaskey) [inline proto/src/determined/run/v1/run.proto:113]: is this set or returned anywhere?

File: master/internal/logretention/logretention.go [hunk 2]
```diff
@@ -95,10 +95,12 @@ func DeleteExpiredTaskLogs(ctx context.Context, days *int16) (int64, error) {
...
-			SELECT task_id, end_time, COALESCE(log_retention_days, %d) AS log_retention_days FROM tasks
-			WHERE task_id IN (SELECT DISTINCT task_id FROM task_logs)
-				AND end_time IS NOT NULL
-		)
+			SELECT COALESCE(r.log_retention_days, %d) as log_retention_days, t.task_id, t.end_time
+			FROM runs as r
+			JOIN run_id_task_id as r_t ON r.id = r_t.run_id
+			JOIN tasks as t ON r_t.task_id = t.task_id
+			WHERE t.end_time IS NOT NULL
+				)
```
Comment (NicholasBlaskey) [inline master/internal/logretention/logretention.go:103]: formatting: this ) isn't aligned to anything

**Review by NicholasBlaskey — APPROVED** [review]
LGTM but I would get @erikwilson to make sure the design look right since I'm not familiar with this feature

File: proto/src/determined/run/v1/run.proto (hunk 1 above)
Comment (salonig23) [reply proto/src/determined/run/v1/run.proto:113]: actually, I don't think so. i'll remove it

**Review by erikwilson — APPROVED** [review]

**Pull request merged**